### ISSUE TITLE
[DDNet server] Detect when the server is being ddosed and freeze all tees

### ISSUE THREAD

**Issue by heinrich5991** [ddnet/ddnet#5357]
converted from #5079.

@CovERUshKA wrote:
> If the server is being ddosed it will freeze all tees.
> When the server is not ddosed anymore it will return all tees to the position before ddos and start timer(3....2.....1....GO). When timer ends it will unfreeze all tees.

**Comment by heinrich5991** [comment]
similar #5080:

@CovERUshKA wrote:
Notify people about the server is being ddosed. Text somewhere or image.
> Add text somewhere or an image to notify people that the server is being ddosed right now. They will faster realize that the server is under ddos and stop moving faster.

**Comment by heinrich5991** [comment]
I don't think freezing all tees is a good idea. This will probably have false positives and cause fails.

Not sure how accurate one can measure whether a server is under attack, but if it can be done somewhat reliably, the notification sounds okay.

**Comment by ChillerDragon** [comment]
Freezing all tees might be a misleading wording here. We probably want to freeze the whole game state. Including switch timers, crazy shotgun bullets etc. There is the pause function already that can be activated by admins.

**Comment by AssassinTee** [comment]
Can the server even detect if it's getting ddossed or are we relying on a third party tool here? I'd likely connect a script with a ddos attack detection tool and pause the server with econ as others suggested. 

> I don't think we could dig through the mess that is the server code and write a proper CGameWorld::Pause() that has no observable effect (e.g. changing the behaviour of subtle bugs relied on by mappers). A warning does sound doable.

I agree with @Learath2, this is likely almost impossible to get perfectly right

Additionally I don't see this as strongly wanted. Do you want the server to pause if a ddos is detected? I can already see a malicious actor start/stopping the game every 30 seconds

**Comment by ChillerDragon** [comment]
> Can the server even detect if it's getting ddossed

Yes the game server can detect traffic spikes. Or maybe even better detect tick delays.

**Comment by AssassinTee** [comment]
> > Can the server even detect if it's getting ddossed
> 
> Yes the game server can detect traffic spikes. Or maybe even better detect tick delays.

Please reopen this, if you think it's the server that should deal with this. I see a lot of downsides to this feature as it might be abused

**Comment by ChillerDragon** [comment]
Yea no I am fine, just wanted to answer your question. Onbgy fng even had a /ddos chat command to get packets per second stats if I remember correctly.
